spec/controllers/contributions_controller.spec.rb

```diff
@@ -0,0 +1,146 @@
+require 'spec_helper'
+
+RSpec.describe Admin::ContributionsController, type: :controller do
+  render_views
+
+  let(:admin) { build_stubbed(:user, :super_admin) }
+  let!(:contribution) { build_stubbed(:contribution) }
+  let(:page) { Capybara::Node::Simple.new(response.body) } 
+
+  let(:invalid_attributes) do
+    { first_name: '' }
+  end
+
+
+  before do
+    allow(controller).to receive(:authenticate_user!)
+    allow(controller).to receive(:current_user).and_return(admin)
+  end
+ 
+  describe 'GET index' do
+    
+    before do
+      get :index
+    end
+
+    it 'renders the index template' do
+      expect(response).to render_template(:index)
+    end
+
+    it 'has http status 200' do
+      expect(response).to have_http_status(:ok)
+    end
+  end
+
+  describe 'GET new' do
+    before do
+      get :new
+    end
+
+    it 'has http status 200' do
+      expect(response).to have_http_status(:ok)
+    end
+
+    it 'assigns the contribution' do
+      expect(assigns(:contribution)).to be_a_new(Contribution)
+    end
+
+    it "should render the form elements" do
+      expect(page).to have_field('contribution[user_id]')
+      expect(page).to have_field('contribution[company_id]')
+      expect(page).to have_field('contribution[repository_id]')
+      expect(page).to have_field('contribution[link]')
+    end
+  end
+
+  describe "POST create" do
+    let(:user) { create(:user) }
+    let(:company) { create(:company) }
+    let(:repository) { create(:repository) }
+
+    let(:contribution_params) {
+      {
+        "link"=>"https://github.com/Codeminer", 
+        "user_id"=>"#{user.id}", 
+        "company_id"=>"#{company.id}",
+        "repository_id" => "#{repository.id}"
+      }
+    }
+
+
+    context "with valid params" do
+      it "creates a new Contribution" do
+        
+        expect {
+          post :create, params: { contribution: contribution_params }
+        }.to change(Contribution, :count).by(1)
+      end
+
+      it "assigns a newly created contribution as @contribution" do
+        post :create, params: { contribution: contribution_params }
+        expect(assigns(:contribution)).to be_a(Contribution)
```

Comment:
use composable matches so you get a better error message in case of error https://www.betterspecs.org/#single